frontera/contrib/scrapy/schedulers/frontier.py

```diff
@@ -72,31 +72,43 @@ def _set_value(self, variable, value):
 
 
 class FronteraScheduler(Scheduler):
+    """
+    Custom Scheduler for Scrapy.
+    Adapts Frontera manager interface to Scrapy.
+
+    Important remarks:
+     - it doesn't enqueue majority of requests produced by middlewares or direct calls to engine, see self.enqueue_request
+       method, and override if needed,
+     - requires SchedulerSpiderMiddleware and SchedulerDownloaderMiddleware.
+    """
 
     def __init__(self, crawler, manager=None):
         self.crawler = crawler
         self.stats_manager = StatsManager(crawler.stats)
         self._pending_requests = deque()
-        self.redirect_enabled = crawler.settings.get('REDIRECT_ENABLED')
         settings = ScrapySettingsAdapter(crawler.settings)
         self.frontier = ScrapyFrontierManager(settings, manager)
         self._delay_on_empty = self.frontier.manager.settings.get('DELAY_ON_EMPTY')
+        self._redirect_enabled = crawler.settings.get('REDIRECT_ENABLED')
         self._delay_next_call = 0.0
-        self.logger = getLogger('frontera.contrib.scrapy.schedulers.FronteraScheduler')
+        self.logger = getLogger('frontera.contrib.scrapy.schedulers.frontier.FronteraScheduler')
 
     @classmethod
     def from_crawler(cls, crawler):
         return cls(crawler)
 
     def enqueue_request(self, request):
```

Comment:
I believe adding middleware which is bypassing the frontier and want's to get it's requests in the local queue requires planning from the user operating the crawler. He would need to understand all the consequences of this. If we take your login example, the user would need to deal with response that comes after previously unseen login request. Frontera custom scheduler will crash on this, because it lacks of ``frontier_request`` in meta. Therefore the middleware which logs in would need to intercept this response or figure out something else.

Frontera already has some ancient mechanism of tagging the requests
https://github.com/scrapinghub/frontera/blob/master/frontera/contrib/scrapy/converters.py#L41
at the moment it's not used.